server/testdriver/core/src/main/java/org/infinispan/server/test/core/AbstractInfinispanServerDriver.java

```diff
@@ -84,22 +91,32 @@ public void prepare(String name) {
          throw new RuntimeException("Failed to create server configuration directory " + confDir);
       }
       URL configurationFileURL;
-      if (new File(configuration.configurationFile()).isAbsolute()) {
+      if (DEFAULT_INFINISPAN_CONFIG_FILES.contains(configuration.configurationFile())) {
+         configurationFileURL = getClass().getClassLoader().getResource(configuration.configurationFile());
+      } else if (new File(configuration.configurationFile()).isAbsolute()) {
          configurationFileURL = Exceptions.unchecked(() -> new File(configuration.configurationFile()).toURI().toURL());
       } else {
          configurationFileURL = getClass().getClassLoader().getResource(configuration.configurationFile());
       }
       if (configurationFileURL == null) {
          throw new RuntimeException("Cannot find test configuration file: " + configuration.configurationFile());
       }
-      Path configurationFilePath;
+
       try {
-         configurationFilePath = Paths.get(configurationFileURL.toURI());
-         // Recursively copy the contents of the directory containing the configuration file to the test target
-         Util.recursiveDirectoryCopy(configurationFilePath.getParent(), confDir.toPath());
+         if (DEFAULT_INFINISPAN_CONFIG_FILES.contains(configuration.configurationFile())) {
```

Comment:
Here you're only copying the configuration file. What happens with the other files the server needs, like `log4j2.xml`?